chatbot-core/rag/retriever/retrieve.py

```diff
@@ -0,0 +1,30 @@
+"""
+Query interface for retrieving the most relevant embedded text chunks using a FAISS vector index.
+"""
+
+from rag.embedding.embedding_utils import load_embedding_model, embed_documents
+from rag.retriever.retriever_utils import load_vector_index, search_index
+
+MODEL_NAME = "sentence-transformers/all-MiniLM-L6-v2"
+
+def get_relevant_documents(query, logger, top_k=5):
```

Comment:
Repeatedly loading the model can be expensive. Allow passing in a preloaded model as a parameter